bitnami/jaeger/templates/migrate-job.yaml

```diff
@@ -0,0 +1,107 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "common.names.fullname" . }}-migrate
+  namespace: {{ .Release.Namespace }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: query
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.migration.annotations "context" $ ) | nindent 4 }}
+    {{- if .Values.commonAnnotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+spec:
+  template:
+    metadata:
+      labels: {{- include "common.labels.standard" . | nindent 8 }}
+        app.kubernetes.io/component: migration
+        {{- if .Values.migration.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.migration.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+      {{- if .Values.migration.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.migration.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
+    spec:
+      restartPolicy: OnFailure
+      {{- if .Values.migration.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.migration.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      initContainers:
+        - name: jaeger-cassandra-schema-grabber
+          image: {{ include "jaeger.image" . }}
+          command: [ "cp", "-T", "-r", "/opt/bitnami/jaeger/cassandra-schema", "/cassandra-schema/" ]
+          volumeMounts:
+            - name: cassandra-schema
+              mountPath: "/cassandra-schema"
+      containers:
+        - name: jaeger-cassandra-migrator
+          image: {{ printf "%s/%s:%s" .Values.cassandra.image.registry .Values.cassandra.image.repository .Values.cassandra.image.tag }}
```

Comment:
I would create a value called `migrationImage` that uses cassandra under the hood. The reason for this is that `cassandra.enabled` will be mandatory in order for the chart to work, and in other charts (check the gitea or appsmith PRs) we allow an existing external database. 

I know this may cause potential mismatches between the Cassandra image in the sub-chart and the one in the chart, however, as we are only using the clqsh client, we don't expect major issues in it and the advantages are huge.

As an alternative, we could add a cassandra-client nami module in the jaeger image but I believe that the required effort does not make it worthwhile

Reply:
What about...?:

```
{{- if not (empty .Values.cassandra.migrationImage) }}
image: {{- include .Values.cassandra.migrationImage . -}}
{{- else }}
image: {{ printf "%s/%s:%s" .Values.cassandra.image.registry .Values.cassandra.image.repository .Values.cassandra.image.tag }}
{{- end }}
```

Reply:
Refined [21fa513](https://github.com/bitnami/charts/pull/13480/commits/21fa5133e83b0bf57f2ef0d1759e72bd718674d8)

Leaving the conversation open and waiting review. Feel free to close it if the solution fits.